### PR TITLE
Feature/adn 759

### PR DESCRIPTION
**Pull request by alebastrov**

### CODE REVIEW

File: oap-ws/src/main/java/oap/ws/WsDeprecated.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+package oap.ws;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target( ElementType.METHOD )
+@Retention( RetentionPolicy.RUNTIME )
+public @interface WsDeprecated {
```
Comment (vovakyrychenko) [inline oap-ws/src/main/java/oap/ws/WsDeprecated.java:34]: use java.lang.Deprecated

File: oap-ws-openapi/src/main/java/oap/ws/openapi/OpenapiGenerator.java [hunk 2]
```diff
@@ -163,21 +183,29 @@ public Result processWebservice( Class clazz, String context ) {
...
-        operation.description( wsDescriptor.description );
-        operation.setRequestBody( prepareRequestBody( params, api ) );
-        operation.setResponses( prepareResponse( returnType, wsDescriptor.produces, api ) );
+        operation.description( wsMethodDescriptor.description );
+        operation.setRequestBody( prepareRequestBody( params ) );
+        operation.setResponses( prepareResponse( returnType, wsMethodDescriptor.produces ) );
+        Optional<WsDeprecated> deprecated = method.findAnnotation( WsDeprecated.class );
+        deprecated.ifPresent( x -> operation.deprecated( true ) );
+        if ( wsSecurityDescriptor != WsSecurityDescriptor.NO_SECURITY_SET ) {
+            operation.addSecurityItem( new SecurityRequirement().addList( "JWT" ) );
```
Comment (vovakyrychenko) [inline oap-ws-openapi/src/main/java/oap/ws/openapi/OpenapiGenerator.java:203]: need to publish permissions from WsSecurity

**Review by vovakyrychenko — CHANGES_REQUESTED** [review]

File: oap-ws/src/main/java/oap/ws/WsDeprecated.java (hunk 1 above)
Comment (alebastrov) [reply oap-ws/src/main/java/oap/ws/WsDeprecated.java:34]: done

File: oap-ws-openapi/src/main/java/oap/ws/openapi/OpenapiGenerator.java (hunk 2 above)
Comment (alebastrov) [reply oap-ws-openapi/src/main/java/oap/ws/openapi/OpenapiGenerator.java:203]: To here:

Method test1
	[GET] /example/test/sort/{id}
	Produces application/json
	Realm param <unsecure>
	Permissions <unsecure>
	Returns String
	Parameters
		id: path String

?

Comment (alebastrov) [reply oap-ws-openapi/src/main/java/oap/ws/openapi/OpenapiGenerator.java]: done

**Pull request merged**